nerfstudio/fields/visibility_field.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 the Regents of the University of California, Nerfstudio Team and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Visibility Field"""
+
+import torch
+from torch import nn
+
+from nerfstudio.cameras.cameras import Cameras
+from nerfstudio.cameras.rays import RaySamples
+
+
+class VisibilityField(nn.Module):
+    """Visibility Field"""
+
+    def __init__(self, cameras: Cameras) -> None:
+        super().__init__()
+        # training camera tranforms
+        # TODO: use optimized cameras
+        self.c2ws = cameras.camera_to_worlds
+        self.c2whs = torch.cat([self.c2ws, torch.zeros_like(self.c2ws[:, :1, :])], dim=1)
+        self.c2whs[:, 3, 3] = 1.0
+        self.w2chs = torch.inverse(self.c2whs)
+        self.K = cameras.get_intrinsics_matrices()
+        self.image_height = cameras.height
+        self.image_width = cameras.width
+
+    @torch.no_grad()
+    def forward(self, ray_samples: RaySamples, camera_chunk_size=50, ray_chunk_size=4096) -> torch.Tensor:
```

Comment:
does it maybe make sense to take in positions (Nx3) directly instead of ray_samples? this would make it compatible with other methods which might want to use visibility like splatting.